src/NzbDrone.Core/MediaFiles/EpisodeFileMovingService.cs

```diff
@@ -1,85 +1,85 @@
-using System;
-using System.Collections.Generic;
 using System.IO;
-using System.Linq;
 using NLog;
 using NzbDrone.Common.Disk;
-using NzbDrone.Common.EnsureThat;
-using NzbDrone.Common.Extensions;
 using NzbDrone.Core.Configuration;
-using NzbDrone.Core.MediaFiles.EpisodeImport;
-using NzbDrone.Core.MediaFiles.Events;
+using NzbDrone.Core.MediaFiles.MediaInfo;
 using NzbDrone.Core.Messaging.Events;
 using NzbDrone.Core.Organizer;
 using NzbDrone.Core.Parser.Model;
-using NzbDrone.Core.Tv;
 
 namespace NzbDrone.Core.MediaFiles
 {
     public interface IMoveEpisodeFiles
     {
-        EpisodeFile MoveEpisodeFile(EpisodeFile episodeFile, Series series);
-        EpisodeFile MoveEpisodeFile(EpisodeFile episodeFile, LocalEpisode localEpisode);
-        EpisodeFile CopyEpisodeFile(EpisodeFile episodeFile, LocalEpisode localEpisode);
+        EpisodeFile MoveEpisodeFile(EpisodeFile episodeFile, LocalEpisode localEpisode, ScriptImportDecisionInfo scriptImportDecisionInfo);
```

Comment:
Most of the information in `ScriptImportDecisionInfo` is already part of `localEpisode`, including `LocalEpisode` itself, there is no need to pass it in to a function as a separate parameter, if more information is necessary then it should be added to `LocalEpisode`.

Reply:
I have been meaning to ask, what is the semantic difference between an `EpisodeFile` and a `LocalEpisode`? I'm asking to see if information such as say the `DownloadClientItem` even belongs into either of those.

Reply:
Going by what I am seeing so far are the `downloadId`(or I could instead replace the `DownloadClientItemClientInfo` with just the `DownloadClientItem` that contains both it and the `downloadId`), which makes a fair amount of sense, and the `OldFiles` which I am not totally sure how much sense it really does make. I would also definitely have to mutate it part way through to add those.

Reply:
`LocalEpisode` is the file at time of import, used to decide whether the file should be imported and hold information about it until it's imported, after which its unused. `EpisodeFile` is what gets stored in the DB and kept going forward.